lib/src/code_generator/global.dart

```diff
@@ -34,20 +36,40 @@ class Global extends LookUpBinding {
           dartDoc: dartDoc,
         );
 
+  List<Typedef>? _typedefDependencies;
+  @override
+  List<Typedef> getTypedefDependencies(Writer w) {
+    if (_typedefDependencies == null) {
+      _typedefDependencies = <Typedef>[];
+
+      // Add typedef's required by the variable's type.
+      final valueType = type.getBaseType();
+      if (valueType.broadType == BroadType.NativeFunction) {
+        _typedefDependencies!.addAll(valueType.nativeFunc!.getDependencies());
+      }
+    }
+    return _typedefDependencies!;
+  }
+
   @override
   BindingString toBindingString(Writer w) {
     final s = StringBuffer();
     final globalVarName = name;
     if (dartDoc != null) {
       s.write(makeDartDoc(dartDoc!));
     }
+    final pointerName = w.wrapperLevelUniqueNamer.makeUnique('_$globalVarName');
+    final dartType = type.getDartType(w);
+    final cType = type.getCType(w);
+    final refOrValue = type.broadType == BroadType.Struct ? 'ref' : 'value';
 
-    final holderVarName =
-        w.wrapperLevelUniqueNamer.makeUnique('_$globalVarName');
     s.write(
-        '${w.ffiLibraryPrefix}.Pointer<${type.getCType(w)}> $holderVarName;\n');
-    s.write(
-        "${type.getDartType(w)} get $globalVarName => ($holderVarName ??= ${w.dylibIdentifier}.lookup<${type.getCType(w)}>('$originalName')).value;\n\n");
+        "late final ffi.Pointer<$dartType> $pointerName = ${w.dylibIdentifier}.lookup<$cType>('$originalName');\n\n");
```

Comment:
`ffi` => `${w.ffiLibraryPrefix}`, this is needed since this prefix will change if some C library has a declaration named `ffi`.